openedx/features/course_experience/static/course_experience/js/WelcomeMessage.js

```diff
@@ -3,18 +3,34 @@ import 'jquery.cookie';
 
 export class WelcomeMessage {  // eslint-disable-line import/prefer-default-export
 
-  constructor(options) {
-    $('.dismiss-message button').click(() => {
-      $.ajax({
-        type: 'POST',
-        url: options.dismissUrl,
-        headers: {
-          'X-CSRFToken': $.cookie('csrftoken'),
-        },
-        success: () => {
-          $('.welcome-message').hide();
-        },
-      });
+  static dismissWelcomeMessage(dismissUrl) {
+    $.ajax({
+      type: 'POST',
+      url: dismissUrl,
+      headers: {
+        'X-CSRFToken': $.cookie('csrftoken'),
+      },
+      success: () => {
+        $('.welcome-message').hide();
+      },
     });
   }
+
+  constructor(options) {
+    // Check to see if the welcome message has been displayed at all.
```

Comment:
Maybe add the following as an intro:
// Dismiss the welcome message if the user clicks dismiss, or auto-dismiss if 
// the user doesn't click dismiss in 7 days from when it was first viewed.